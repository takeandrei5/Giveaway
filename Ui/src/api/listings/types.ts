import { PaginatedResult } from '../../utils/types';

export type ItemData = { id: string; image: string; title: string; createdAt: Date };

export type CreateListingRequest = {
	title: string;
	description: string;
	category: number;
	images: string[];
};

export type UpdateListingRequest = {
	title: string;
	description: string;
	category: number;
	images: string[];
};

export interface FetchListingsResponse {
	listings: PaginatedResult<ItemData>;
}

export type ListingInformation = {
	id: string;
	title: string;
	description: string;
	category: number;
	createdAt: Date;
	images: string[];
};

export type OwnerInformation = {
	email: string;
	name: string;
	image: string;
};

export type FetchListingDetailsResponse = { listingInfo: ListingInformation } & { ownerInfo: OwnerInformation };
