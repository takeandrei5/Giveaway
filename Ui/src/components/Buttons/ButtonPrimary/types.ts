import { ButtonProps } from '@chakra-ui/react';

export type ButtonPrimaryProps = Omit<ButtonProps, 'color'>;
