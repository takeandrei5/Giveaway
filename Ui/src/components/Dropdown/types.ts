export type DropdownProps = {
	id: string;
	options: DropdownOption[];
	onChangeHandler: (value: string) => void;
	name: string;
	value: string | number | readonly string[];
	isInvalid?: false;
};

export type DropdownOption = {
	value: string | number;
	displayValue: string;
};
