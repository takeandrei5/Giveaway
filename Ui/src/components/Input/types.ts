export type InputProps = {
	id: string;
	name: string;
	placeholder: string;
	value?: string;
	onBlur?: (e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) => void;
	onChange?: (e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) => void;
	onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	onKeyPress?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	onKeyUp?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	disabled?: boolean;
	multiline?: boolean;
	isInvalid?: boolean;
	height?: string | number;
	label?: string;
	leftIcon?: JSX.Element;
	rightIcon?: JSX.Element;
	rows?: number;
	width?: string | number;
};
