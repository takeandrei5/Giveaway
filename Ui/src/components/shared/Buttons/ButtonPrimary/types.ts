import { ButtonProps } from '@chakra-ui/react';

export type ButtonPrimaryProps = ButtonProps;
