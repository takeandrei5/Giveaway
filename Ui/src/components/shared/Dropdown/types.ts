export type DropdownProps = {
	options: DropdownOption[];
	onChangeHandler: (value: string) => void;
};

export type DropdownOption = {
	value: string | number;
	displayValue: string;
};
