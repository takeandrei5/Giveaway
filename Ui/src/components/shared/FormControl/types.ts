export type FormControlProps = {
	children: React.ReactElement;
	label: string;
	id: string;
	name: string;
};

export type useCheckFormIsInvalidProps = {
	isInvalid: boolean;
};
