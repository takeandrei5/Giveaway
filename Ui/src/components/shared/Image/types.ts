import { ImageProps as NextImageProps } from 'next/image';

export type ImageProps = NextImageProps & {
	height: string;
	width: string;
	backgroundColor?: string;
	borderRadius?: string;
	draggable?: boolean;
};
