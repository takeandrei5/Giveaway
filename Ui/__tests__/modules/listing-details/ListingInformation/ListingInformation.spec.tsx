import { ListingInformation } from '@modules/listing-details/ListingInformation';
import { ListingInformationProps } from '@modules/listing-details/ListingInformation/types';
import { render } from '@testing-library/react';

jest.mock('dateformat', () => jest.fn());

describe('ListingInformation', () => {
	it('should match snapshot', () => {
		// Arrange
		const props: ListingInformationProps = {
			title: 'test-title',
			description: 'test-description',
			createdAt: new Date(),
			images: ['test-image-url'],
		};

		// Act
		const { container } = render(<ListingInformation {...props} />);

		// Assert
		expect(container).toMatchSnapshot();
	});
});
