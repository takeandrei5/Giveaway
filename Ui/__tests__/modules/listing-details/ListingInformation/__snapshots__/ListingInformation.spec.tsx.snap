// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingInformation should match snapshot 1`] = `
<div>
  <div
    class="css-1jkzbrs"
  >
    <span
      class="chakra-text css-wfx7tx"
      data-testid="typography"
    >
      test-title
    </span>
    <span
      class="chakra-text css-1jv3994"
      data-testid="typography"
    >
      test-description
    </span>
    <div
      class="css-8xl60i"
    >
      <span
        class="chakra-text css-jp1ymq"
        data-testid="typography"
      >
        Added on: 
      </span>
    </div>
  </div>
</div>
`;
